moped-editor/src/views/dev/LookupsView/index.js

```diff
@@ -31,6 +40,52 @@ const LookupsView = () => {
     fetchPolicy: "no-cache",
   });
 
+  const [selectedRecordKey, setSelectedRecordKey] = useState(null);
+
+  /**
+   * We're using history here (and elsewhere) because it's not possible to use react-router
+   * to replace window.loctation w/o forcing a re-render.
+   * See // https://github.com/remix-run/react-router/issues/8908
+   */
+  const history = createBrowserHistory();
+
+  /**
+   * Create a ref index for every table element on the page so that we can scroll to them
+   * */
+  const refs = useMemo(
+    () =>
+      SETTINGS.reduce((prev, recordType) => {
+        prev[recordType.key] = createRef();
+        return prev;
+      }, {}),
+    []
+  );
+
+  /**
+   * Use the record hash from the URL, if present. This only happens on page load.
+   * */
+  const { hash: recordKeyHash } = useLocation();
+  useEffect(() => {
+    if (!recordKeyHash) {
+      return;
+    }
+    const recordKey = recordKeyHash.replace("#", "").replace("-", "_");
+    setSelectedRecordKey(recordKey);
+  }, [recordKeyHash]);
+
+  /**
+   * Listens for changes to the selected record key and scrolls to its table
+   */
+  useEffect(() => {
+    if (!selectedRecordKey) {
+      return;
+    }
+    const ref = refs?.[selectedRecordKey];
+    if (ref?.current) {
+      ref.current.scrollIntoView({ behavior: "smooth" });
```

Comment:
I noticed that, if I paste `https://localhost:3000/moped/dev/lookups#moped-milestones` into a new tab, the scroll to the milestones section doesn't happen as expected.

I added `console.log("here is that ref", ref, ref?.current);` between lines 83 & 84 and ref.current remains `null` so the scroll never fires.

I think it might be the fact that [updates to `ref.current` don't trigger a re-render](https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/), but I also noticed that removing the dependency array on line 61 from the `useMemo` fixes this (even though that doesn't seem like the right fix). I'm going to think about it some more.

Reply:
ah, fascinating. now i'm seeing a probable race condition b/t when we set ` selectedRecordKey` from the URL and when those refs are hydrated with `.current` after rendering. 

the underlying problem, i think, is that we should be using the `useLocation()` hook to have a stateful reference to the current browser path, but things get really wonky when you start using  the react-router method (`navigate` irrc) to push to history. 

anyway, looking forward to getting back to this later today. thanks Mike!

Reply:
ok, i _think_ what's happening is that first effect hook fires on first mount, but the `loading` is blocking the ref'd table components from rendering. so hence the `null` value in `ref.current`. i'm not sure why this was working locally for me—`loading` is always `true` on the first render.

in any case, i moved the loading spinner down into the record tables, and the ref'd tabled components should always render on the first mount.

i'm open to other ideas!

Reply:
oh, i also added `loading` as a dependency to that first hook. i haven't seen it happen, but page might not scroll far enough if it's scrolling to empty tables. better to wait until we have data to scroll.

Reply:
This makes sense to me - thanks for the explanation! 🙏